smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/MemberShapeEncodeXMLGenerator.kt

```diff
@@ -81,9 +82,14 @@ abstract class MemberShapeEncodeXMLGenerator(
                     }
                 }
                 is TimestampShape -> {
-                    val format = determineTimestampFormat(nestedMember, nestedMemberTarget, defaultTimestampFormat)
-                    val encodeValue = "${ClientRuntimeTypes.Serde.TimestampWrapper}($nestedMemberTargetName, format: .$format), forKey: ${ClientRuntimeTypes.Serde.Key}(\"${nestedMemberResolvedName}\")"
-                    writer.write("try $containerName.encode($encodeValue)")
+                    val codingKey = "${ClientRuntimeTypes.Serde.Key}(\"${nestedMemberResolvedName}\")"
+                    val code = TimestampEncodeGenerator(
+                        containerName,
+                        nestedMemberTargetName,
+                        codingKey,
+                        TimestampHelpers.getTimestampFormat(nestedMember, nestedMemberTarget, defaultTimestampFormat)
+                    ).generate()
+                    writer.write(code)
```

Comment:
could you make use of formatters here?

also applies to rest of the similar code snippets, potentially the generate method takes the writer as param and uses formatter inside it.